documentation/src/test/java/example/AutoCloseDemo.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package example;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.junit.jupiter.api.AutoClose;
+import org.junit.jupiter.api.Test;
+
+// tag::user_guide_example[]
+class AutoCloseDemo {
+
+	@AutoClose
+	Connection connection = DriverManager.getConnection("jdbc:mysql://localhost/testdb");
+
+	@Test
+	void usersTableHasEntries() throws SQLException {
+		ResultSet resultSet = connection.createStatement().executeQuery("SELECT * FROM users");
+
+		assertTrue(resultSet.next());
+	}
+
+	AutoCloseDemo() throws SQLException {
+	}
```

Comment:
No need for that constructor.

Reply:
Unfortunately `DriverManager.getConnection(url)` throws a checked SQLException and the used ctor + throws allows writing the given test.
Maybe I can elaborate a better example.

Reply:
Ahhh... OK. The reason wasn't immediately clear in the GitHub UI, but I get it now.

How about this instead?

```java
class AutoCloseDemo {

	@AutoClose
	Connection connection = getJdbcConnection("jdbc:mysql://localhost/testdb");

	@Test
	void usersTableHasEntries() throws SQLException {
		ResultSet resultSet = connection.createStatement().executeQuery("SELECT * FROM users");

		assertTrue(resultSet.next());
	}

	private static Connection getJdbcConnection(String url) {
		try {
			return DriverManager.getConnection(url);
		}
		catch (SQLException ex) {
			throw new RuntimeException(ex);
		}
	}

}
```

That at least makes the code clear when viewed in the User Guide.

Also, we could _hide_ the implementation of `getJdbcConnection()` in the User Guide.

Reply:
Parameterless `Constructor() throws Exception { }` is kind a "pattern" to allow writing field declarations easily without catching all the thrown exceptions, having extracting them into methods or put everything in setup methods which declare `throws Exception` instead.

Reply:
Yes, I am aware of the language feature; however, the point of an example in the User Guide is to demonstrate a specific feature _without_ superfluous noise. 

For this feature, our goal is to demonstrate how to use `@AutoClose` on a field whose type should be closed. The reader does not need to be concerned with any checked exceptions thrown by the code that we choose to use to create such a field.

If you don't want to make the changes, that's fine. I can address this while merging the PR.

In any case, my current plan is to move `getConnection()` to a `JdbcUtils` class (which is not shown in the example) and invoke that via `JdbcUtils.getConnection(...)`.

This is somewhat analogous to the `WebClient` examples we have, where we do not show the implementation of `WebClient`. So I might even refactor the `@AutoClose` example to use `WebClient` instead of a JDBC `Connection` and point out that the `WebClient` has a `close()` method that needs to be invoked. Anyway, I'm just brainstorming, but hopefully you get the point.

Reply:
FWIW, the `Statement` and `ResultSet` in the current example also need to be closed.

In light of that, we definitely need to rework the example.

Reply:
I just noticed you already incorporated the `getJdbcConnection()` approach. Thanks.

Reply:
> FWIW, the `Statement` and `ResultSet` in the current example also need to be closed.

I'm quite sure that the statement and resultset are also closed when the owning connection is closed and makes this a concise unit test. But there are definitely better examples for the user guide e.g. your proposed WebClient example.
I just want this PR merged that we can get rid of our own AutoCloseExtension ;)
Merry Christmas!

Reply:
I went with `WebClient`.

You can see the result in my [branch](https://github.com/junit-team/junit5/compare/main...sbrannen:junit5:issues/3592-AutoClose).